cirq-superstaq/cirq_superstaq/service.py

```diff
@@ -165,6 +173,46 @@ def _resolve_target(self, target: Union[str, None]) -> str:
 
         return target
 
+    def _validate_cirq_circuit(self, circuits: Union[cirq.Circuit, Sequence[cirq.Circuit]]) -> None:
+        """Validates that the input is either a single cirq.Circuit or a list of cirq.Circuit
+        instances.
+
+        Args:
+            circuit: The circuit to run.
+
+        Raises:
+            InvalidInputError: If the 'circuits' input is not a cirq.Circuit or a list of
+            cirq.Circuit instances.
+            InvalidInputError: If the 'circuits' input is a list containing elements that
+            are not cirq.Circuit instances.
+        """
+        if not isinstance(circuits, (cirq.Circuit, Sequence)):
+            raise InvalidInputError(
+                "Invalid 'circuits' input. Must be a cirq.Circuit or a sequence of cirq.Circuit"
+                "instances."
+            )
+        if isinstance(circuits, Sequence) and not all(
+            isinstance(circuit, cirq.Circuit) for circuit in circuits
+        ):
+            raise InvalidInputError(
+                "Invalid 'circuits' input. All elements in the sequence must be cirq.Circuit"
+                "instances."
+            )
+
+    def _validate_get_counts(self, repetitions: int) -> None:
+        """Validates that the number of repetitions is non-negative.
+
+        Args:
+            repetitions: The number of times to run the circuit.
+
+        Raises:
+            InvalidInputError: If input repetitions is negative.
+        """
+        if repetitions < 0:
```

Comment:
is there any use case for `repetitions=0`? if not,
```suggestion
        if repetitions <= 0:
```
(and replace "non-negative" with "positive" in docstrings/error message)

Reply:
also might be worth checking that it's an integer (or at least can safely be cast as one)? e.g.
```python
if repetitions != int(repetitions):
    raise ....